packages/cds-odata-annotation-converter/src/transforms/annotation/flattened.ts

```diff
@@ -144,30 +159,48 @@ function convertToExpandedStructure(
             i += 2;
             continue;
         }
-
-        const flatProperty: Element = createElementNode({
-            name: Edm.PropertyValue,
-            range: propertyRange,
-            contentRange: propertyRange,
-            attributes: {
-                [Edm.Property]: createPropertyAttribute(segment.value, segment.range)
+        if (segment.value === ReservedProperties.Type && value?.type === STRING_LITERAL_TYPE) {
+            const flatProperty: Element = createElementNode({
+                name: Edm.Record,
+                attributes: {
+                    [Edm.Type]: createAttributeNode(Edm.Type, value.value, segment.range, nodeRange(value, false))
+                },
+                range: propertyRange
+            });
+            if (expandedStructure.length > 0) {
+                const last = expandedStructure[i - 1];
+                last.element.contentRange = createRange(segment.range?.start, valueRange?.end);
```

Comment:
We already do this in line 59, I don't think we need to do it here as well.

Reply:
removed